execution_engine_testing/tests/src/test/system_contracts/auction/mod.rs

```diff
@@ -171,26 +201,73 @@ fn should_support_contract_staking() {
             .query(None, delegation_key, &[])
             .expect("should have delegation bid")
     {
-        // assert_ne!(
-        //     delegator.staked_amount(),
-        //     delegate_amount,
-        //     "staked amount should execeed delegation amount due to rewards"
-        // );
         delegator.staked_amount()
     } else {
         U512::zero()
     };
 
-    // unstake from contract
+    // restake from contract
     builder
         .exec(
             ExecuteRequestBuilder::contract_call_by_name(
                 account,
                 &contract_name,
-                &unstake,
+                &entry_point_name,
                 runtime_args! {
+                    ARG_ACTION => restake,
                     ARG_AMOUNT => increased_delegate_amount,
                     ARG_VALIDATOR => validator_pk.clone(),
+                    ARG_NEW_VALIDATOR => other_validator_pk.clone()
+                },
+            )
+            .build(),
+        )
+        .commit()
+        .expect_success();
+
+    assert!(
+        builder.query(None, delegation_key, &[]).is_err(),
+        "delegation record should be removed"
+    );
+
+    assert_eq!(
+        post_delegation_balance,
+        builder.get_purse_balance(contract_purse),
+        "at this point, unstaked token has not been returned"
+    );
+
+    for _ in 0..=7 {
+        // crank era
+        builder.run_auction(timestamp_millis, vec![]);
+    }
+
+    let delegation_key = Key::BidAddr(BidAddr::DelegatedPurse {
+        validator: other_validator_pk.to_account_hash(),
+        delegator: contract_purse.addr(),
+    });
+
+    if let StoredValue::BidKind(BidKind::Delegator(delegator)) = builder
```

Comment:
We don't fail if it's not a `StoredValue::BidKind(BidKind::Delegator())` ?

Reply:
It would bottom out pretty quickly, but fair point. I will explicitly assert over it in next commit.